spring-kafka/src/main/java/org/springframework/kafka/core/StreamsBuilderFactoryBean.java

```diff
@@ -137,6 +158,7 @@ public void setCloseTimeout(int closeTimeout) {
 
 	@Override
 	protected StreamsBuilder createInstance() throws Exception {
+		Assert.notNull(this.streamsConfig, "'streamsConfig' must not be null");
```

Comment:
You know, Soby, looking at this one more time I retract my previous requirement.

Looks like we are indeed are good to live with the state without config until the first `start()` and that really may happen somewhere late by logic.

So, yes, let's check for the `streamsConfig` if `autoStartup == true` and one more check in the `start()` as you had before.

Make sense?

Sorry for the noise.